### PR TITLE
Add macOS build docs

### PR DESCRIPTION
**Pull request by andkerr**
Download and Build Instructions for MacOS (Intel Hardware), with Git and Homebrew

## Goal of the PR
This PR adds Git/Homebrew download and build instructions for Mac users getting started with Minetest. It mainly incorporates commands from the `macos.yml` workflow file located in the `.github/workflows` directory, which are what I used to create an inital build-from-source.

## How does the PR work?
See above, incorporates existing build commands from the MacOS workflow file.

The MacOS instructions don't include a minimum dependency version tables right now. I found this useful when looking at the GNU/Linux instructions, but don't know how to figure out minimum versions without more knowledge of how Minetest works.

## Does it resolve any reported issue?
N/A

## Does this relate to a goal in [the roadmap](../doc/direction.md)?
Will more accessible build instructions bring more Mac users to Minetest?

## If not a bug fix, why is this PR needed? What usecases does it solve?
Minetest's README currently includes no download/build instructions for MacOS users, and the commands contained in the MacOS Github workflow that _do_ work are hard to find.

## To do

This PR is Ready for Review.

- [ ] Determine minimum version requirements (I have no experience with this, and would welcome advice or contibutions)

## How to test
Ideally, a few (Intel) Mac users would run these instructions top to bottom, and verify that they generate a successful build like they did on my machine.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -386,6 +386,61 @@ Build the binaries as described above, but make sure you unselect `RUN_IN_PLACE`
...
+
+Download Minetest's fork of Irrlicht:
+
+```
+git clone --depth 1 https://github.com/minetest/irrlicht.git lib/irrlichtmt
+```
+
+#### Build
+
+```
```
Comment (sfan5) [inline README.md:424]: for syntax highlighting
```suggestion
```bash
```

**Review by sfan5 — APPROVED** [review]
I don't have a Mac to test but these steps match what I remember to be correct, so should be fine.

File: README.md (hunk 1 above)
Comment (andkerr) [reply README.md]: Great, I just incorporated your suggestion.

File: README.md [hunk 2]
```diff
@@ -386,6 +386,61 @@ Build the binaries as described above, but make sure you unselect `RUN_IN_PLACE`
 Open the generated project file with Visual Studio. Right-click **Package** and choose **Generate**.
 It may take some minutes to generate the installer.
 
+### Compiling on MacOS (Intel Architecture)
```
Comment (sfan5) [inline README.md:389]: In theory there is nothing preventing this from working on ARM so I'd rather not mention this until we know for sure that it *doesn't* work there.

Comment (andkerr) [reply README.md:389]: Sure thing, fixed

**Comment by rubenwardy** [comment]
@Jordach

**Review by rubenwardy — APPROVED** [review]

**Pull request merged**